metastats-domain/src/main/groovy/life/qbic/metastats/filter/SequencingModeCalculator.groovy

```diff
@@ -14,33 +12,31 @@ import org.apache.logging.log4j.Logger
  */
 class SequencingModeCalculator {
 
-    private static final Logger LOG = LogManager.getLogger(SequencingModeCalculator.class)
-
     /**
      * Calculates the sequencing mode based on all filenames of a sample.
      * If the files contain R1 and R2 then the mode is paired end. Else it is single end.
      * @param filename a string containing all filenames of a sample comma separated
-     * @return String of determined sequencing mode 
+     * @return String of determined sequencing mode
      */
-    static String calculateSequencingMode(String filename) {
+    static String calculateSequencingMode(String filenames) {
         String sequencingMode = ""
-        String[] files = filename.split(",")
+        String[] files = filenames.split(",")
```

Comment:
The current approach of splitting a concatenated String is very error-prone. I would suggest extracting this step from this method or if possible to remove the concatenation and splitting of strings from this information transfer completely.

Reply:
#7